File: frag-nav/src/test/java/com/ncapdevi/fragnav/FragNavTransactionOptionsTest.java
package com.ncapdevi.fragnav;

import android.support.annotation.AnimRes;
import android.support.annotation.StyleRes;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.util.Pair;
import android.view.View;

import org.junit.Test;

import static org.junit.Assert.assertTrue;

/**
 * Created by niccapdevila on 2/15/17.
 */

public class FragNavTransactionOptionsTest {

    @Test
    public void buildTransactionOptions() {
        String breadCrumbShortTitle = "Short Title";
        String breadCrumbTitle = "Long Title";

        @AnimRes
        int enterAnim = 1;
        @AnimRes
        int exitAnim = 2;
        @AnimRes
        int popEnterAnim = 3;
        @AnimRes
        int popExitAnim = 4;

        @StyleRes
        int transitionStyle = 5;

        FragNavTransactionOptions fragNavTransactionOptions = FragNavTransactionOptions.Companion.newBuilder()
                .breadCrumbShortTitle(breadCrumbShortTitle)
                .breadCrumbTitle(breadCrumbTitle)
                .transition(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
                .customAnimations(enterAnim, exitAnim, popEnterAnim, popExitAnim)
                .transitionStyle(transitionStyle)
                .addSharedElement(new Pair<View, String>(null, "test"))
                .addSharedElement(new Pair<View, String>(null, "test2")).build();

        assertTrue(breadCrumbShortTitle.equalsIgnoreCase(fragNavTransactionOptions.getBreadCrumbShortTitle()));
        assertTrue(breadCrumbTitle.equalsIgnoreCase(fragNavTransactionOptions.getBreadCrumbTitle()));

        assertTrue(transitionStyle == fragNavTransactionOptions.getTransitionStyle());

        assertTrue(FragmentTransaction.TRANSIT_FRAGMENT_FADE == fragNavTransactionOptions.getTransition());


        assertTrue(enterAnim == fragNavTransactionOptions.getEnterAnimation());
        assertTrue(exitAnim == fragNavTransactionOptions.getExitAnimation());
        assertTrue(popEnterAnim == fragNavTransactionOptions.getPopEnterAnimation());
        assertTrue(popExitAnim == fragNavTransactionOptions.getPopExitAnimation());


        assertTrue(fragNavTransactionOptions.getSharedElements().size() == 2);


    }
}

File: frag-nav/src/test/java/com/ncapdevi/fragnav/FragNavTransactionOptionsTest.kt
package com.ncapdevi.fragnav

import androidx.annotation.AnimRes
import androidx.annotation.StyleRes
import androidx.fragment.app.FragmentTransaction
import com.nhaarman.mockitokotlin2.mock
import org.junit.Assert.assertTrue
import org.junit.Test

/**
 * Created by niccapdevila on 2/15/17.
 */

class FragNavTransactionOptionsTest {

    @Test
    fun buildTransactionOptions() {
        val breadCrumbShortTitle = "Short Title"
        val breadCrumbTitle = "Long Title"

        @AnimRes
        val enterAnim = 1
        @AnimRes
        val exitAnim = 2
        @AnimRes
        val popEnterAnim = 3
        @AnimRes
        val popExitAnim = 4

        @StyleRes
        val transitionStyle = 5

        val fragNavTransactionOptions = FragNavTransactionOptions.newBuilder()
            .breadCrumbShortTitle(breadCrumbShortTitle)
            .breadCrumbTitle(breadCrumbTitle)
            .transition(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
            .customAnimations(enterAnim, exitAnim, popEnterAnim, popExitAnim)
            .transitionStyle(transitionStyle)
            .addSharedElement(Pair(mock(), "test"))
            .addSharedElement(Pair(mock(), "test2")).build()

        assertTrue(breadCrumbShortTitle.equals(fragNavTransactionOptions.breadCrumbShortTitle, ignoreCase = true))
        assertTrue(breadCrumbTitle.equals(fragNavTransactionOptions.breadCrumbTitle, ignoreCase = true))

        assertTrue(transitionStyle == fragNavTransactionOptions.transitionStyle)

        assertTrue(FragmentTransaction.TRANSIT_FRAGMENT_FADE == fragNavTransactionOptions.transition)


        assertTrue(enterAnim == fragNavTransactionOptions.enterAnimation)
        assertTrue(exitAnim == fragNavTransactionOptions.exitAnimation)
        assertTrue(popEnterAnim == fragNavTransactionOptions.popEnterAnimation)
        assertTrue(popExitAnim == fragNavTransactionOptions.popExitAnimation)


        assertTrue(fragNavTransactionOptions.sharedElements.size == 2)
    }
}
